### PR TITLE
add: Linux kernels

### PR DESCRIPTION
**Pull request by Chadano**
Add 5 new pacscripts, four of which install parts of the Linux kernel from the Ubuntu mainline repositories. The fifth one installs all 4 parts as pacdeps in one singular package.

### CODE REVIEW

**Review by Henryws — CHANGES_REQUESTED** [review]
Add your email to the `maintainer` variable.
Fix your PR title (change to `add:`). There are no features.
On the linux-kernel-deb pacscript, you **must** have a Deb to install, not just pacdeps.

**Comment by Chadano** [comment]
> Add your email to the `maintainer` variable. Fix your PR title (change to `add:`). There are no features. On the linux-kernel-deb pacscript, you **must** have a Deb to install, not just pacdeps.

1. I don't have an email I am currently comfortable sharing, can I use matrix in-place instead?
2. Wizard set the title, I will set it to add now.
3. Is just "linux-kernel" good?

**Comment by Henryws** [comment]
linux-kernel should work.
You must include an email. If you don't have one, you could use https://anonaddy.me

**Comment by Chadano** [comment]
> linux-kernel should work. You must include an email. If you don't have one, you could use https://anonaddy.me

I will just create one with one of my various domains, will be 1 sec.

File: packages/linux-headers-deb/linux-headers-deb.pacscript [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name="linux-headers-deb"
+gives="linux-headers-deb"
```
Comment (Henryws) [inline packages/linux-headers-deb/linux-headers-deb.pacscript:2]: gives is what the package name is.
```suggestion
gives="linux-headers"
```

Do the same for every Deb pacscript.

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/linux-headers-deb/linux-headers-deb.pacscript (hunk 1 above)
Comment (Chadano) [reply packages/linux-headers-deb/linux-headers-deb.pacscript:2]: Doing that now, will be a moment.

**Comment by Chadano** [comment]
I believe I have resolved all of the issues.

**Comment by Chadano** [comment]
> I believe I have resolved all of the issues.

Will I need to re-edit the package list?

**Comment by Henryws** [comment]
No, `gives` is what the package actually is. For most purposes, its $name - deb

**Comment by Chadano** [comment]
> No, `gives` is what the package actually is. For most purposes, its $name - deb

I see.

**Comment by Chadano** [comment]
@Henryws So rather than linux-image-unsigned it should be the full deb name without the .deb?

**Comment by Henryws** [comment]
If you were to install the Deb, the package would be named something like, `linux-image-unsigned`, not `linux-image-unsigned-deb`. The `-deb` is a pacstall convention to differentiate package types. `gives` purpose is to explicitly name what the package is known as by apt after installation.

**Comment by Chadano** [comment]
> If you were to install the Deb, the package would be named something like, `linux-image-unsigned`, not `linux-image-unsigned-deb`. The `-deb` is a pacstall convention to differentiate package types. `gives` purpose is to explicitly name what the package is known as by apt after installation.

I see, can you clarify one thing for me?

`linux-headers-6.0.0-060000_6.0.0-060000.202210022231_all.deb` or `linux-headers-6.0.0-060000_6.0.0-060000.202210022231_all`

**Comment by Henryws** [comment]
For what?

**Comment by Chadano** [comment]
For the gives section, does it need the .deb

**Comment by Henryws** [comment]
No, it's $name minus the `-deb` suffix

**Comment by Chadano** [comment]
> No, it's $name minus the `-deb` suffix

By $name do you mean debian name or name of the apt package?

File: packages/linux-kernel/linux-kernel.pacscript [hunk 2]
```diff
@@ -0,0 +1,5 @@
+name="linux-kernel"
+version="6.0"
+pacdeps=("linux-headers-deb" "linux-headers-generic-deb" "linux-image-unsigned-deb" "linux-modules-deb")
+description="Provides the latest Linux kernel via the Ubuntu mainline repositories"
+maintainer="Chadano <chadano@llamawebdesign.co.uk>"
```
Comment (Henryws) [inline packages/linux-kernel/linux-kernel.pacscript:5]: you must include a `url` variable. It could just be an empty file, but it must be included.

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/linux-kernel/linux-kernel.pacscript (hunk 2 above)
Comment (Chadano) [reply packages/linux-kernel/linux-kernel.pacscript:5]: so just
url=""
prepare=""
build=""

File: packages/linux-kernel/linux-kernel.pacscript [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name="linux-kernel"
+version="6.0"
+url="https://www.learningcontainer.com/wp-content/uploads/2020/05/sample-zip-file.zip"
+pacdeps=("linux-headers-deb" "linux-headers-generic-deb" "linux-image-unsigned-deb" "linux-modules-deb")
```
Comment (Henryws) [inline packages/linux-kernel/linux-kernel.pacscript:4]:
```suggestion
pacdeps=("linux-headers-deb" "linux-headers-generic-deb" "linux-modules-deb" "linux-image-unsigned-deb")
```

**Review by Henryws — CHANGES_REQUESTED** [review]

**Review by Henryws — APPROVED** [review]

**Pull request merged**